src/ServiceManagement/Profile/Commands.Profile/ServicePrincipal/NewAzureSubscriptionServicePrincipal.cs

```diff
@@ -0,0 +1,57 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Management.Automation;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using Microsoft.WindowsAzure.Management;
+using Microsoft.WindowsAzure.Management.Models;
+using static Microsoft.WindowsAzure.Management.Models.SubscriptionServicePrincipalListResponse;
+
+namespace Microsoft.WindowsAzure.Commands.Profile
+{
+
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, AzureSubscriptionServicePrincipalNounName)]
```

Comment:
Cmdlet must implement SHoudlProcess

Reply:
I am not able to see any precedent of this. Existing cmdlets like RemoveAzureSubscriptionCommand etc do not override ShouldProcess. Why is this necessary?